app/javascript/utils/Languages.js

```diff
@@ -0,0 +1,97 @@
+import LANGUAGES from '../data/minifiedLanguages.json'
+import supportedLanguages from '../data/supportedLanguages'
+import _ from 'lodash'
+
+/**
+ * Tries to match the input to a language in the system.
+ * It accepts isoCodes or Display Names
+ * If unable to match it, it returns null
+ * @param {Object} potentialCodeorLanguage
+ * @return {Object}
+ */
+function tryToMatchLanguage (potentialCodeorLanguage) {
+  if (!potentialCodeorLanguage) return null
+  let matchedLang = LANGUAGES.find(o => o.c === potentialCodeorLanguage);
+  if (!matchedLang) {
+    matchedLang = LANGUAGES.find(o => o.d === potentialCodeorLanguage);
+  }
+  return matchedLang
+}
+
+/**
+ * Returns an object containing all the contact-method booleans
+ * for a language. The values default to false, unless specified true in
+ * 'supportedLanguages.js'
+ * @param {Object} language - must be in the form of {c: isoCode, d: displayName}
+ * @return {Object}
+ */
+function getLanguageSupported(language = {c:null, d:null}) {
+  let supportedLanguageReference = supportedLanguages.find(y => y.name === language.d)
+  if (supportedLanguageReference) {
+    supportedLanguageReference["code"] = language.c
+  } else {
+    supportedLanguageReference = {
+      "name": language.d,
+      "code": language.c,
+      "supported": {
+        "sms": false,
+        "email": false,
+        "phone": false
+      }
+    }
+  }
+  return supportedLanguageReference
+}
+
+/**
+ * Returns a grouped array of all Languages in the system, formatted to work
+ * with react-select (AKA in the format { value: isoCode, label: displayName }).
+ * The groups are by whether the language is Supported and Unsupported.
+ * These options are all alphabetized as well.
+ * @return {Array} groupedOptions
+ */
+function getLanguagesAsOptions () {
```

Comment:
I really like how this looks!  Question though, do we think we need the * on unsupported languages when they're in that section? Do we think we need it at all now that the warning shows up? might be something to run by the PO team

Reply:
Good question! I'll follow up with them and let you know.

Reply:
Per PO Team, the asterisk should remain unchanged